Pagerduty Provider Plugin Crashed

### Terraform Version
1.7.5

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

Stack trace from the terraform-provider-pagerduty_v3.11.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 487 [running]:
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.expandIncidentUrgencyType(...)
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_service.go:772
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.expandIncidentUrgencyRule({0xf0ee40?, 0xc00074e558?})
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_service.go:741 +0x4d4
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.buildServiceStruct(0xc000532320?)
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_service.go:439 +0x805
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.resourcePagerDutyServiceUpdate(0xc000998200, {0xf8c460?, 0xc000532320?})
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_service.go:529 +0x58
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x1375c40?, {0x1375c40?, 0xc0000c86c0?}, 0xd?, {0xf8c460?, 0xc000532320?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:800 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000358e00, {0x1375c40, 0xc0000c86c0}, 0xc000992d00, 0xc000998000, {0xf8c460, 0xc000532320})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:919 +0x845
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00012f980, {0x1375c40?, 0xc0000c8540?}, 0xc000b34be0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x1375b98?, {0x1375c40?, 0xc0000c81e0?}, 0xc000b34be0)
	github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0003921e0, {0x1375c40?, 0xc000394990?}, 0xc000266070)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10a4bc0?, 0xc0003921e0}, {0x1375c40, 0xc000394990}, 0xc000540000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:503 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001c9000, {0x1375c40, 0xc000394900}, {0x137b038, 0xc000583040}, 0xc00079c000, 0xc00043e120, 0x1a3ce38, 0x0)
	google.golang.org/grpc@v1.60.1/server.go:1372 +0xe49
google.golang.org/grpc.(*Server).handleStream(0xc0001c9000, {0x137b038, 0xc000583040}, 0xc00079c000)
	google.golang.org/grpc@v1.60.1/server.go:1783 +0x1031
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.60.1/server.go:1016 +0x68
created by google.golang.org/grpc.(*Server).serveStreams.func2
	google.golang.org/grpc@v1.60.1/server.go:1027 +0x12e

--- a/pagerduty/resource_pagerduty_service.go
+++ b/pagerduty/resource_pagerduty_service.go
@@ -769,8 +769,15 @@ func flattenIncidentUrgencyRule(v *pagerduty.IncidentUrgencyRule) []interface{} 
 }
 
 func expandIncidentUrgencyType(v interface{}) *pagerduty.IncidentUrgencyType {
-	riut := v.([]interface{})[0].(map[string]interface{})
 	incidentUrgencyType := &pagerduty.IncidentUrgencyType{}
+	riut := make(map[string]interface{})
+
+	data, ok := v.([]interface{})
+	if ok && len(data) > 0 && !isNilFunc(data[0]) {
+		riut = data[0].(map[string]interface{})
+	} else {
+		return incidentUrgencyType
+	}
 
 	if v, ok := riut["type"]; ok {
 		incidentUrgencyType.Type = v.(string)

